cmdstanpy/model.py

```diff
@@ -330,25 +334,48 @@ def format_model(
             )
 
             if canonicalize:
-                if isinstance(canonicalize, str):
-                    cmd.append('--canonicalize=' + canonicalize)
-                elif isinstance(canonicalize, Iterable):
-                    cmd.append('--canonicalize=' + ','.join(canonicalize))
+                if cmdstan_version_before(2, 29):
+                    if isinstance(canonicalize, bool):
+                        cmd.append('--print-canonical')
+                    else:
+                        raise ValueError(
+                            "Invalid arguments passed for current CmdStan"
+                            + " version({})\n".format(
+                                cmdstan_version() or "Unknown"
+                            )
+                            + "--canonicalize requires 2.29 or higher"
+                        )
                 else:
-                    cmd.append('--print-canonical')
-
+                    if isinstance(canonicalize, str):
+                        cmd.append('--canonicalize=' + canonicalize)
+                    elif isinstance(canonicalize, Iterable):
+                        cmd.append('--canonicalize=' + ','.join(canonicalize))
+                    else:
+                        cmd.append('--print-canonical')
+
+            # before 2.29, having both --print-canonical
+            # and --auto-format printed twice
             if not (cmdstan_version_before(2, 29) and canonicalize):
                 cmd.append('--auto-format')
 
+            if not cmdstan_version_before(2, 29):
+                cmd.append(f'--max-line-length={max_line_length}')
+            elif max_line_length != 78:
+                raise ValueError(
+                    "Invalid arguments passed for current CmdStan version"
+                    + " ({})\n".format(cmdstan_version() or "Unknown")
+                    + "--max-line-length requires 2.29 or higher"
+                )
+
             out = subprocess.run(
                 cmd, capture_output=True, text=True, check=True
             )
             if out.stderr:
                 get_logger().warning(out.stderr)
             result = out.stdout
-            if save:
+            if overwrite_file:
                 if result:
-                    if not unsafe:
+                    if backup:
                         shutil.copyfile(self.stan_file, self.stan_file + '.bak')
```

Comment:
I'm splitting hairs here, but multiple call will overwrite the older backup file?

Reply:
Yes. We could add a random component to the string or a timestamp if we wanted

Reply:
Maybe timestamp?

Reply:
Done

Reply:
I see the wisdom of doing this.

the potential downside is lots of backup files.  since Stan program files aren't that big, this is fine, other than visual clutter when listing a directory.   were someone to build some kind of workflow tool around CmdStanPy, this behavoir would be most useful.

Reply:
The docstring/doc hints at the better way of doing this, which is storing your files in something like Git and then using `backup=False` to avoid the clutter. Either way, I imagine most people will probably only run this one or two times per model, probably when they see the deprecation warning, and then never again